### DISCUSSION TITLE
Activity_per_s.mat - TimeDate - Activity

### DISCUSSION THREAD

**Discussion by anerjolo** [Ergo-Tools/ActiPASS#97] [Help and Q&A (processing, quality checking or statistics)] [answered]
Hello everyone,

I work as a researcher at the University of Jyväskylä, and I am particularly interested in ActiPASS's ability to detect stair walking. I am currently working with the "Activity_per_s.mat" file and had a couple of questions regarding the data format. I couldn't find this information in the documentation.

Time Representation in the "TimeDate" Variable: I noticed that the values in the "TimeDate" variable begin with 739582 in my data. Do you have information on what time representation format is used in this variable? Specifically, I would like to understand what this number corresponds to and how it can be interpreted in terms of the actual date and time (e.g., UNIX).

Activity Column Values: I would also appreciate clarification on what the numeric values in the "Activity" column represent. Do you have information on whether you provide a description of the different activities or actions these numbers correspond to?

**Comment by pasan-h** [comment, marked as answer]
Hi, welcome to ActiPASS community 😀.
The "Activity_per_s.mat" file is considered an intermediate file and hence not documented. We encourage end-users to use the variables from [documented daily (24h) tables, or interval-based tables](https://github.com/Ergo-Tools/ActiPASS/wiki/ActiPASS-variables-and-other-quality-check-files) whenever possible. These contain very important quality-check data which is not available on high resolution 1s epoch table (Activity_per_s.mat). The quality checking information (various flags and visualizations) are specially designed in such a way the end-user understand the capabilities and limitations of ActiPASS. 

Is there a specific reason that you want to use the 1s classifications? If so the best way of understanding is to have a look at ActiPASS stats/variable generation code at https://github.com/Ergo-Tools/ActiPASS/tree/main/stats. 

To answer your specific question, the time data is just a MATLAB datenumber, and the numbers (which corresponds to different behaviours) can be found at https://github.com/Ergo-Tools/ActiPASS/blob/main/stats/genVariables.m

**Reply by anerjolo** [reply]
Hi,

Thank you so much! :)

After a long wait, I finally got the analysis working, and I really appreciate your help in clarifying that it was a MATLAB datenumber. Huge thanks for that!

I'm particularly interested in the 1s classification because it helps identify different activities, such as stair climbing.

**Comment by pasan-h** [comment]
Hello, great! Anyway, ActiPASS outputs all behaviours in the standard tables including stair-walking. I still do not understand why 1s data should be used. Do you want to study temporal patterns (and transitions) of different behaviours? 
**IF SO:**
I should warn you that we have to interpret this 1s epoch table very carefully. The effective epoch length of this time-series actually vary from around 2s to 20s. If we include lying and sleep effective epoch is prolonged as high as 30 minutes. Opposite is also true. Exact transitions of some activities does not make sense in the real world but do occur algorithmically. In order to handle activity transitions and durations we need to use the concepts of bouts but [allowing certain amount of breaks and thresholds](https://github.com/Ergo-Tools/ActiPASS/wiki/How-bout-variables-are-derived-by-ActiPASS#how-physical-activity-pa-sedentary-behaviour-sb-or-intensity-class-sed-lpa-mvpa-vpa-etc-bouts-are-derived-by-actipass).
For an example if you use 1s output and derive a parameter such as **median duration of stair-walking segments** you can get results which do not make much sense in real world.
I do not imply that Acti4/ActiPASS algorithms are wrong. They work well (as described in validation studies). However finding temporal patterns from a time-series of activities should be done with enough understanding of underlying algorithms and human perceptions (what actually we want to find?).

**Reply by anerjolo** [reply]
Thank you for your detailed response. I’m particularly interested in strength-demanding activities, and for older adults, stair walking is one of them. ActiPass seems to be one of the first tools to analyze stair walking on a larger scale, so I would be interested in using it to analyze our own datasets.

I'm specifically interested in **1-second data because, with proper labeling and timestamps, it could theoretically be possible to examine whether stair walking intensity, or at least the duration of stair walking bouts, could be assessed in some way**. I fully understand that there is always uncertainty in free-living physical activity monitoring, and this must be critically considered when interpreting data.

Have you considered whether stair walking intensity could be assessed from the data in some way?

Thanks again for your help!

**Reply by pasan-h** [reply]
> Have you considered whether stair walking intensity could be assessed from the data in some way?

We get cadence during stair-walking.

**Reply by anerjolo** [reply]
Hi,

Thank you again for your response :)

We would be interested in exploring the intensity of stair walking in free-living environment. Is it possible to extract stair walking cadence from the analyses in any way? I haven’t found this in the documentation yet.

Thank you very much!

**Reply by pasan-h** [reply]
Cadence is in the same 1s table (data structure). These features are not documented because they are not properly validated or can only be used in limited scenarios influenced by a lot of other factors. To summarise it's the sole responsibility of the end-user for subsequent analysis or intepretation of this feature and to make sure its validity. 

Cadence is given for each second (but derived from a roughly 5 second window). But it make sense to calculate the cadence for each bout of stair walking by summing the cadence values and dividing by the duration of the corresponding bout. 
However see my warnings [above ](https://github.com/Ergo-Tools/ActiPASS/discussions/97#discussioncomment-12136042)and [below](https://github.com/Ergo-Tools/ActiPASS/discussions/97#discussioncomment-12166732). Unless you have enough power, I do not reccomend using time of stair-walking or cadence in studies. The article I linked below has enough power because it's a large epi study.

**Reply by anerjolo** [reply]
Hi,

Thank you very much for the information. I understand the matter and appreciate the clarification!

**Comment by pasan-h** [comment]
Hi,
Yes, I too think it's easier to use the 1-s data file in your situation. Anyway ActiPASS can also generate the same variables described in the wiki based on an specific time-intervals. Here we need to use an additional diary with time-stamps of each interval. 

Anyway some hints/warnings you can consider 😀.

1. ActiPASS automatic reference position (individual calibration) is based on periods oof walking present in free living data. If the ref-position go wrong, you can either get too much or too few stair-walking (as well as misclassifications of walking as stair-walking). Stair-walking is the most sensitive to ref-positions. People who are mostly sedentary have the worst ref-positions. 
2. Stair-walking is actually all stair-walking like activities including hiking, inclined-walking (depend on inclination). 
3. In general, stair-walking based on auto ref-positions are good enough for big epi-studies. But for small cohorts, we need to use manual individual calibration (ref-positions based on diary time-stamps).
4. For an epi-study which delas with stair-walking and also some supplementary info about performance, check our ProPASS article:
https://link.springer.com/article/10.1007/s00125-024-06090-y

**Comment by toteitto** [comment]
Hi, thanks again for the helpful explanations. I have an additional question about the Activity_per_s.mat file and thought it could fit well into this thread

I noticed that there is also an SVM variable in the 1 s data. Could you clarify how this SVM is calculated? Is it the standard Signal Vector Magnitude (e.g. based on the 3-axis acceleration), or something different within ActiPASS? 

Also, is the SVM reported second-by-second, or is it somehow smoothed or derived from a window (similar to the activity classification, where the effective epoch length can be longer than 1 second)?

Thanks a lot!

**Comment by pasan-h** [comment]
SVM provided in the 1s table is filtered multiple times (more suitable as a generic indicator of movements rather than the VM of three axes). and perhaps not suitable for your needs. It's first found in the [FindAnglesAndVM.m](https://github.com/Ergo-Tools/ActiPASS/blob/main/library/FindAnglesAndVM.m) function. See line number 54. SVM is found after first 6Hz low-pass filter. Then another round of filtration (2s moving mean) happens before saving the value to disk..
See the [CLI workflow](https://github.com/Ergo-Tools/ActiPASS/blob/main/cli/actipass_cli.m) lines 461 and 521.
